src/Glpi/Inventory/Asset/VirtualMachine.php

```diff
@@ -341,9 +357,29 @@ protected function createVmComputer()
                     if ($this->conf->states_id_default != '-1') {
                         $input['states_id'] = $this->conf->states_id_default;
                     }
-                    $datarules = $rule->processAllRules($input);
+                    $datarules = $rule->processAllRules($input, [], ['class' => $input['itemtype'], 'return' => true]);
                     if (isset($datarules['_no_rule_matches']) && ($datarules['_no_rule_matches'] == '1') || isset($datarules['found_inventories'])) {
                         $computervm->update($input);
+
+                        if (isset($datarules['found_inventories'])) {
+                            $rulesmatched = new \RuleMatchedLog();
+                            $agents_id = $this->agent->fields['id'];
+                            if (empty($agents_id)) {
+                                $agents_id = 0;
+                            }
+                            $inputrulelog = [
+                                'date'      => date('Y-m-d H:i:s'),
+                                'rules_id'  => $datarules['rules_id'],
+                                'items_id'  => $computers_vm_id,
+                                'itemtype'  => $input['itemtype'],
+                                'agents_id' => $agents_id,
+                                'method'    => 'inventory'
+                            ];
+                            $rulesmatched->add($inputrulelog, [], false);
+                        }
+                    } else {
+                        //refused by rules
+                        continue;
```

Comment:
Seems useless.
```suggestion
```

Reply:
It's required, otherwise it will continue executing the function inside the ``foreach`` and will attempt to create/update machine components, etc. which shouldn't happen if asset update has been denied by rules.